mm2src/adex_cli/src/scenarios/init_mm2_cfg.rs

```diff
@@ -0,0 +1,302 @@
+        self.inquire_gui()?;
+        self.inquire_net_id()?;
+        self.inquire_passphrase()?;
+        self.inquire_allow_weak_password()?;
+        self.inquire_rpc_password()?;
+        self.inquire_userhome()?;
+        self.inquire_dbdir()?;
+        self.inquire_rpcip()?;
+        self.inquire_rpcport()?;
+        self.inquire_rpc_local_only()?;
+        self.inquire_i_am_a_seed()?;
+        self.inquire_seednodes()?;
+        self.inquire_hd_account_id()?;
+        Ok(())
+    }
+
+    fn inquire_dbdir(&mut self) -> Result<(), ()> {
+        let is_reachable_dir = |dbdir: &InquireOption<String>| -> Result<Validation, CustomUserError> {
+            match dbdir {
+                InquireOption::None => Ok(Validation::Valid),
+                InquireOption::Some(dbdir) => {
+                    let path = Path::new(dbdir);
+                    if path.is_dir().not() {
+                        return Ok(Validation::Invalid(
+                            format!("\"{dbdir}\" - is not a directory or does not exist").into(),
+                        ));
+                    }
+                    Ok(Validation::Valid)
+                },
+            }
+        };
+
+        self.dbdir = CustomType::<InquireOption<String>>::new("What is dbdir")
+                .with_placeholder(DEFAULT_OPTION_PLACEHOLDER)
+                .with_help_message("AtomicDEX API database path. Optional, defaults to a subfolder named DB in the path of your mm2 binary")
+                .with_validator(is_reachable_dir)
+                .prompt()
+                .map_err(|error| {
+                    error!("Failed to get dbdir: {error}");
+                })?.into();
+
+        Ok(())
+    }
+
+    fn inquire_gui(&mut self) -> Result<(), ()> {
+        self.gui = Text::new("What is the client identifier, gui:")
+                .with_default(DEFAULT_GID)
+                .with_placeholder(DEFAULT_GID)
+                .with_help_message("Information about your GUI; place essential info about your application (name, version, etc.) here. For example: AtomicDEX iOS 1.0.1")
+                .prompt()
+                .map_err(|error| {
+                    error!("Failed to get gui: {error}");
+                })?.into();
+        Ok(())
+    }
+
+    fn inquire_net_id(&mut self) -> Result<(), ()> {
+        self.net_id = CustomType::<u16>::new("What is the network `mm2` is going to be a part, net_id:")
+                .with_default(DEFAULT_NET_ID)
+                .with_help_message(r#"Nework ID number, telling the AtomicDEX API which network to join. 7777 is the current main network, though alternative netids can be used for testing or "private" trades"#)
+                .with_placeholder(format!("{DEFAULT_NET_ID}").as_str())
+                .prompt()
+                .map_err(|error| {
+                    error!("Failed to get net_id: {error}");
+                })?.into();
+        Ok(())
+    }
+
+    fn inquire_passphrase(&mut self) -> Result<(), ()> {
```

Comment:
If a `default_password` is used, does the user get to see this default generated password?

Reply:
Yes, the get to see